pkg/apis/config/feature_flags.go

```diff
@@ -30,6 +30,15 @@ const (
 	StableAPIFields = "stable"
 	// AlphaAPIFields is the value used for "enable-api-fields" when alpha APIs should be usable as well.
 	AlphaAPIFields = "alpha"
+	// FullEmbeddedStatus is the value used for "embedded-status" when the full statuses of TaskRuns and Runs should be
+	// embedded in PipelineRunStatusFields, but ChildReferences should not be used.
+	FullEmbeddedStatus = "full"
+	// BothEmbeddedStatus is the value used for "embedded-status" when full embedded statuses of TaskRuns and Runs as
+	// well as ChildReferences should be used in PipelineRunStatusFields.
+	BothEmbeddedStatus = "both"
```

Comment:
cc @jerop and @lbernick for that. =)

Reply:
I went back to look at the API WG notes where this was discussed, and it just says "To prevent being broken between versions" :'( The TEP states: "Users can opt in to use `both` at any time, but it is never the default value. It provides a seamless transition for API clients for a short period needed to upgrade to minimal." 
The use case in mind here is if someone wants to update their installation of pipelines and also their use of client libraries: first they would swap to "both", then swap to using ChildRefs, then swap to "minimal".